File: java/common-support/src/main/java/net/multi/language/common/message/event/order/OrderNotChargedEvent.java
package net.multi.language.common.message.event.order;

import net.multi.language.common.marker.message.Channel;
import net.multi.language.common.marker.message.MessageDetails;
import net.multi.language.common.marker.message.Service;
import net.multi.language.common.message.AbstractDomainEvent;

import java.util.Objects;

@MessageDetails(
        publisher = Service.BILLING_SERVICE,
        subscribers = Service.ORDER_SERVICE,
        channel = Channel.ORDER_NOT_CHARGED
)
public class OrderNotChargedEvent extends AbstractDomainEvent {

    private static final String NAME = "OrderNotCharged";

    private final long customerId;
    private final long orderId;
    private final String reason;

    public OrderNotChargedEvent(final long messageId, final long eventId, final long customerId, long orderId,
                                final String reason) {
        super(messageId, eventId);
        this.customerId = customerId;
        this.orderId = orderId;
        this.reason = reason;
    }

    @Override
    public String getName() {
        return NAME;
    }

    public long getCustomerId() {
        return customerId;
    }

    public long getOrderId() {
        return orderId;
    }

    public String getReason() {
        return reason;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        OrderNotChargedEvent that = (OrderNotChargedEvent) o;
        return customerId == that.customerId &&
                orderId == that.orderId &&
                Objects.equals(reason, that.reason);
    }

    @Override
    public int hashCode() {
        return Objects.hash(customerId, orderId, reason);
    }
}

File: python/src/common/message/event/order/order_not_charged_event.py
from typing import ClassVar

from common.message.abstract_domain_event import AbstractDomainEvent


class OrderNotChargedEvent(AbstractDomainEvent):
    NAME: ClassVar[str] = "OrderNotCharged"

    customer_id: int
    order_id: int
    reason: str

    def get_name(self) -> str:
        return self.NAME

    def __eq__(self, other) -> bool:
        if not isinstance(other, OrderNotChargedEvent):
            return False
        return (
            self.customer_id == other.customer_id
            and self.order_id == other.order_id
            and self.reason == other.reason
        )

    def __hash__(self) -> int:
        return hash((self.customer_id, self.order_id, self.reason))
